### PR TITLE
[BugFix] Prohibit creating partitions in advance

### LINKED ISSUE #27122: [auto create list partition] Create partitions in advance, the result is null
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE TABLE `duplicate_table_without_null` ( `k1`  date not null, `k2`  datetime not null,`k3`  char(20) not null, `k4`  varchar(20) not null, `k5`  string not null, `k6`  boolean not null, `k7`  tinyint not null, `k8`  smallint not null, `k9`  int not null, `k10`  bigint not null, `k11` largeint not null, `k12` decimal(27,9) not null, `k13` int not null) DUPLICATE KEY(`k1`, `k2`, `k3`, `k4`, `k5`) PARTITION BY (k7) (PARTITION p1 VALUES IN (("2022-04-01")), PARTITION p2 VALUES IN (("2022-04-02")), PARTITION p3 VALUES IN (("2022-04-03"))) DISTRIBUTED BY HASH(`k1`, `k2`, `k3`) PROPERTIES (  "replication_num" = "3", "storage_format" = "v2");
show partitions from duplicate_table_without_null order by PartitionName;

CREATE TABLE `duplicate_table_without_null2` ( `k1`  date not null, `k2`  datetime not null,`k3`  char(20) not null, `k4`  varchar(20) not null, `k5`  string not null, `k6`  boolean not null, `k7`  tinyint not null, `k8`  smallint not null, `k9`  int not null, `k10`  bigint not null, `k11` largeint not null, `k12` decimal(27,9) not null, `k13` int not null) DUPLICATE KEY(`k1`, `k2`, `k3`, `k4`, `k5`) PARTITION BY (k7) (PARTITION p1 VALUES IN (("1")), PARTITION p2 VALUES IN (("2")), PARTITION p3 VALUES IN (("3"))) DISTRIBUTED BY HASH(`k1`, `k2`, `k3`) PROPERTIES (  "replication_num" = "3", "storage_format" = "v2");
show partitions from duplicate_table_without_null2 order by PartitionName;
```
image

### Expected behavior (Required)
1.partition values is incorrect, failed
2.Create partitions in advance, the result is not null

### Real behavior (Required)
1.partition values is incorrect, successfully
2.Create partitions in advance, the result is null

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
mysql> select current_version();
+--------------------+
| current_version()  |
+--------------------+
| branch-3.1-72fb1c1 |
+--------------------+
1 row in set (0.00 sec)

### PR DESCRIPTION
**Pull request by Astralidea**
Fixes #27122

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|2|50.00%|[702]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -698,6 +698,9 @@ private PartitionDesc getPartitionDesc(StarRocksParser.PartitionDescContext cont
         } else {
             // For hive/iceberg/hudi partition & automatic partition
             ListPartitionDesc listPartitionDesc = new ListPartitionDesc(columnList, partitionDescList);
+            if (context.listPartitionDesc().size() > 0) {
```
[the remark this answers is not in the document]
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:701]: you are right.

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|2|50.00%|[700]|

**Pull request merged**